src/prefect/cli/deployment.py

```diff
@@ -383,6 +394,9 @@ async def set_schedule(
         except json.JSONDecodeError:
             updated_schedule = RRuleSchedule(rrule=rrule_string, timezone=timezone)
 
+    if no_schedule:
+        updated_schedule = NoSchedule()
```

Comment:
Instead of creating a new object to represent the absence of a schedule, you can use `None`. The API will understand that to mean no schedule and behave accordingly.
```suggestion
        updated_schedule = None
```